src/Symfony/Component/Mime/NamedAddress.php

```diff
@@ -39,4 +41,25 @@ public function toString(): string
     {
         return $this->getEncodedNamedAddress();
     }
+
+    public static function fromString(string $string): self
+    {
+        if (false === strpos($string, '<')) {
+            return new static($string, $string);
+        }
+
+        $matched = preg_match(
+            static::FROM_STRING_PATTERN,
+            $string,
+            $matches
+        );
+        if (!$matched) {
+            throw new \RuntimeException(sprintf('Could not parse (%s) to a named address', $string));
+        }
+
+        return new static(
```

Comment:
I suppose that we expect a encoded address, so we must decode the parts here, right?

Reply:
You mean the inverse of `Rfc2231Encoder`?

Reply:
I was more thinking about IDN. Not sure if this is needed though.

Reply:
As `NamedAddressTest::testConstructor` shows you can also provide an un-encoded address.